### PR TITLE
Fix `df.quantile` on all missing data

### LINKED ISSUE #2792: dataframe.quantile on empty all missing data raises
```python
In [4]: import dask.dataframe as dd; import pandas as pd; import numpy as np

In [5]: df = pd.DataFrame({"A": []})

In [6]: ddf = dd.from_pandas(df, 2)

In [7]: df.quantile(0.5)
Out[7]:
A   NaN
Name: 0.5, dtype: float64

In [8]: ddf.quantile(0.5)
Out[8]:
Dask Series Structure:
npartitions=1
A    float64
A        ...
Name: 0.5, dtype: float64
Dask Name: quantiles-concat, 7 tasks

In [9]: ddf.quantile(0.5).compute()
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-9-4bed07fb63b8> in <module>()
----> 1 ddf.quantile(0.5).compute()

~/Envs/dask-dev/lib/python3.6/site-packages/dask/dask/base.py in compute(self, **kwargs)
    136         dask.base.compute
    137         """
--> 138         (result,) = compute(self, traverse=False, **kwargs)
    139         return result
    140

~/Envs/dask-dev/lib/python3.6/site-packages/dask/dask/base.py in compute(*args, **kwargs)
    334     postcomputes = [a.__dask_postcompute__() if is_dask_collection(a)
    335                     else (None, a) for a in args]
--> 336     results = get(dsk, keys, **kwargs)
    337     results_iter = iter(results)
    338     return tuple(a if f is None else f(next(results_iter), *a)

~/Envs/dask-dev/lib/python3.6/site-packages/dask/dask/threaded.py in get(dsk, result, cache, num_workers, **kwargs)
     73     results = get_async(pool.apply_async, len(pool._pool), dsk, result,
     74                         cache=cache, get_id=_thread_get_id,
---> 75                         pack_exception=pack_exception, **kwargs)
     76
     77     # Cleanup pools associated to dead threads

~/Envs/dask-dev/lib/python3.6/site-packages/dask/dask/local.py in get_async(apply_async, num_workers, dsk, result, cache, get_id, rerun_exceptions_locally, pack_exception, raise_exception, callbacks, dumps, loads, **kwargs)
    519                         _execute_task(task, data)  # Re-execute locally
    520                     else:
--> 521                         raise_exception(exc, tb)
    522                 res, worker_id = loads(res_info)
    523                 state['cache'][key] = res

~/Envs/dask-dev/lib/python3.6/site-packages/dask/dask/compatibility.py in reraise(exc, tb)
     58         if exc.__traceback__ is not tb:
     59             raise exc.with_traceback(tb)
---> 60         raise exc
     61
     62 else:

~/Envs/dask-dev/lib/python3.6/site-packages/dask/dask/local.py in execute_task(key, task_info, dumps, loads, get_id, pack_exception)
    288     try:
    289         task, data = loads(task_info)
--> 290         result = _execute_task(task, data)
    291         id = get_id()
    292         result = dumps((result, id))

~/Envs/dask-dev/lib/python3.6/site-packages/dask/dask/local.py in _execute_task(arg, cache, dsk)
    268     elif istask(arg):
    269         func, args = arg[0], arg[1:]
--> 270         args2 = [_execute_task(a, cache) for a in args]
    271         return func(*args2)
    272     elif not ishashable(arg):

~/Envs/dask-dev/lib/python3.6/site-packages/dask/dask/local.py in <listcomp>(.0)
    268     elif istask(arg):
    269         func, args = arg[0], arg[1:]
--> 270         args2 = [_execute_task(a, cache) for a in args]
    271         return func(*args2)
    272     elif not ishashable(arg):

~/Envs/dask-dev/lib/python3.6/site-packages/dask/dask/local.py in _execute_task(arg, cache, dsk)
    269         func, args = arg[0], arg[1:]
    270         args2 = [_execute_task(a, cache) for a in args]
--> 271         return func(*args2)
    272     elif not ishashable(arg):
    273         return arg

~/Envs/dask-dev/lib/python3.6/site-packages/dask/dask/array/percentile.py in merge_percentiles(finalq, qs, vals, Ns, interpolation)
     96     L = list(zip(*[(q, val, N) for q, val, N in zip(qs, vals, Ns) if N]))
     97     if not L:
---> 98         raise ValueError("No non-trivial arrays found")
     99     qs, vals, Ns = L
    100

ValueError: No non-trivial arrays found
```

### PR DESCRIPTION
**Pull request by jsignell**
- [x] Closes #2792
- [x] Tests added / passed
- [x] Passes `black dask` / `flake8 dask` / `isort dask`

### CODE REVIEW

**Comment by jsignell** [comment]
rerun tests

**Comment by jsignell** [comment]
This change is pretty weird, but it's such a niche situation that it feels ok to me. What do you think @jcrist or @jrbourbeau?

File: dask/dataframe/tests/test_dataframe.py [hunk 1]
```diff
@@ -1408,6 +1408,14 @@ def test_quantile_tiny_partitions():
     assert r == 2
 
 
+def test_quantile_trivial_partitions():
+    """See https://github.com/dask/dask/issues/2792"""
+    df = pd.DataFrame({"A": []})
```
Comment (jcrist) [inline dask/dataframe/tests/test_dataframe.py:1413]: Can you also add a test for all `nan` values as well?

File: dask/array/percentile.py [hunk 2]
```diff
@@ -207,14 +209,16 @@ def merge_percentiles(finalq, qs, vals, interpolation="lower", Ns=None):
 
     L = list(zip(*[(q, val, N) for q, val, N in zip(qs, vals, Ns) if N]))
     if not L:
-        raise ValueError("No non-trivial arrays found")
+        if raise_on_nan:
+            raise ValueError("No non-trivial arrays found")
+        return np.full((len(qs[0]) - 2,), np.nan)
```
Comment (jcrist) [inline dask/array/percentile.py:214]: Mild legibility nit, feel free to ignore if you disagree:

```suggestion
        return np.full(len(qs[0]) - 2, np.nan)
```

**Review by jcrist — COMMENTED** [review]
Overall this seems fine to me, just a few tiny comments.

File: dask/array/percentile.py (hunk 2 above)
Comment (jsignell) [reply dask/array/percentile.py:214]: nah that is good.

**Comment by jsignell** [comment]
The failures are probably related to github action being a little iffy this morning.

**Pull request merged**